src/v2i-hub/CARMACloudPlugin/src/CARMACloudPlugin.cpp

```diff
@@ -623,11 +624,10 @@ QByteArray CARMACloudPlugin::UncompressBytes(const QByteArray compressedBytes) c
             char buffer[BUFFER_SIZE] = {0};
             strm.avail_out = BUFFER_SIZE;
             strm.next_out = (Byte *)buffer;
-			//Uncompress finished
-            isDone = inflate(&strm, Z_FINISH);
-            outBuf.append(buffer);
-        } while (Z_STREAM_END != isDone); //Reach the end of stream to be uncompressed 
-    }else{
+            isDone = inflate(&strm, Z_NO_FLUSH);
+            outBuf.append(buffer, BUFFER_SIZE - strm.avail_out);
+		} while (Z_STREAM_END != isDone); // Reach the end of stream to be uncompressed
+	}else{
 		PLOG(logWARNING) << "Error initalize stream. Err code = " << err << std::endl;
```

Comment:
Do we still want to execute the last line of this method if this error happens?

Reply:
It seems useful to know that the stream is not readable and what the actual error is. What do you suggest?

Reply:
I am asking whether we want the statements on L634 and L635 to execute in this error case. I have no problem with the logging except if it is an error we may want to make it error level and not warning? Just curious whether we want to stop doing everything in this case or not.

Reply:
Updated it to error type and handled it silently by ignoring it for now.